Sources/Moya/Plugins/NetworkLoggerPlugin.swift

```diff
@@ -1,135 +1,197 @@
         }
 
         return output
     }
 
-    func logNetworkResponse(_ response: Response, target: TargetType) -> [String] {
+    func logNetworkResponse(_ response: Response, target: TargetType, isFromError: Bool) -> [String] {
+
+        //Response presence check
         guard let httpResponse = response.response else {
-            return [format(loggerId, date: date, identifier: "Response", message: "Received empty network response for \(target).")]
+            return [newEntry(identifier: "Response", message: "Received empty network response for \(target).")]
         }
 
+        // Adding log entries for each given log option
         var output = [String]()
 
-        output += [format(loggerId, date: date, identifier: "Response", message: httpResponse.description)]
+        output.append(newEntry(identifier: "Response", message: httpResponse.description))
 
-        if isVerbose, let stringData = String(data: responseDataFormatter?(response.data) ?? response.data, encoding: String.Encoding.utf8) {
-            output += ["Body: " + stringData]
+        if (isFromError && configuration.errorResponseLoggingOptions.contains(.body))
+            || configuration.successResponseLoggingOptions.contains(.body) {
+
+            let stringOutput = configuration.responseDataFormatter(response.data)
+            output.append(newEntry(identifier: "Response Body", message: stringOutput))
         }
 
         return output
-      }
+    }
 
-      func logNetworkError(_ error: MoyaError, target: TargetType) -> [String] {
-          //Some errors will still have a response, like errors due to Alamofire's HTTP code validation.
-          if let moyaResponse = error.response {
-              return logNetworkResponse(moyaResponse, target: target)
-          }
 
-          //Errors without an HTTPURLResponse are those due to connectivity, time-out and such.
-          return [format(loggerId, date: date, identifier: "Error", message: "Error calling \(target) : \(error)")]
-      }
+    func logNetworkError(_ error: MoyaError, target: TargetType) -> [String] {
+        //Some errors will still have a response, like errors due to Alamofire's HTTP code validation.
+        if let moyaResponse = error.response {
+            return logNetworkResponse(moyaResponse, target: target, isFromError: true)
+        }
+
+        //Errors without an HTTPURLResponse are those due to connectivity, time-out and such.
+        return [newEntry(identifier: "Error", message: "Error calling \(target) : \(error)")]
+    }
 }
 
-fileprivate extension NetworkLoggerPlugin {
-    static func reversedPrint(_ separator: String, terminator: String, target: TargetType, items: Any...) {
-        for item in items {
-            print(item, separator: separator, terminator: terminator)
+//MARK: - Configuration
+public extension NetworkLoggerPlugin {
+    struct Configuration {
+
+        public typealias OutputType = (_ target: TargetType, _ items: [Any]) -> Void
```

Comment:
`[Any]` was previously `Any...`
Some questions here:
- the previous logger actually sent strings wrapped in an array. I do not understand why. Feel free to enlight me.
- why `Any` and not simply `String`? Entries sent by the logger are always `String`.

Reply:
if all entries are strings, feel free to replace the `Any` to `String` and see if it builds! if so, we can probably ship it as well

Reply:
i've replaced Any with String and it works fine :)